### PR TITLE
Add support for retention periods

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: archiver.py [hunk 1]
```diff
@@ -35,15 +35,16 @@
 ))
 
 
-def filter_files(files, days_to_ignore):
-    """Filter files based on the date in the filename
+def filter_by_age(files, comparitor):
```
Comment (richvdh) [inline archiver.py]: comparator

File: archiver.py [hunk 2]
```diff
@@ -177,6 +181,30 @@ def progress_cb(bytes_downloaded, total_size):
         sftp.close()
         client.close()
 
+        # We now go and delete any files that are older than the retention
+        # period, if specified
+        if service.retention_period_days:
+            service_base_dir = os.path.join(self.base_dir, service.name)
```
Comment (richvdh) [inline archiver.py]: erm; it feels odd that we are handling all hosts here whereas a service is host-specific. (What if I wanted to keep logs from hera for 6 months but juno for 1 week?).

And yes, I know that will leave the stuff from old hosts hanging around. Just go and clear them out manually.

File: archiver.py [hunk 3]
```diff
@@ -35,15 +35,16 @@
...
 
-def filter_files(files, days_to_ignore):
-    """Filter files based on the date in the filename
+def filter_by_age(files, comparitor):
+    """Filter files based on the date in their name relative to today.
 
     Args:
-        files (list(str))
-        days_to_ignore(int): Ignore the last N days of logs
+        files (list(str)): List of filenames with a date in them
```
Comment (richvdh) [inline archiver.py]: any iterable will do - doesn't have to be a list?

**Review by richvdh — CHANGES_REQUESTED** [review]

File: archiver.py (hunk 2 above)
Comment (erikjohnston) [reply archiver.py]: Oh, good catch

Comment (erikjohnston) [reply archiver.py]: Though I'm not sure I want to change the config, as it doesn't currently have any host specific options

File: archiver.py [hunk 4]
```diff
@@ -39,7 +39,7 @@ def filter_by_age(files, comparator):
     """Filter files based on the date in their name relative to today.
 
     Args:
-        files (list(str)): List of filenames with a date in them
+        files (iterable(str)): List of filenames with a date in them
```
Comment (richvdh) [inline archiver.py]: could probably have done with `s/List of //` here, but nm

**Review by richvdh — APPROVED** [review]
lgtm

**Pull request merged**